src/nvim/os/env.c

```diff
@@ -135,7 +135,20 @@ int os_setenv(const char *name, const char *value, int overwrite)
   }
 #endif
   uv_mutex_lock(&mutex);
-  int r = uv_os_setenv(name, value);
+  int r;
+#ifdef WIN32
+  // LC_ALL, LANGUAGE, LANG and LC_MESSAGES must be obtained using getenv()
+  // in libintl, so we must use _putenv_s().
+  if (striequal(name, "LC_ALL") || striequal(name, "LANGUAGE")
+      || striequal(name, "LANG") || striequal(name, "LC_MESSAGES")) {
+    r = _putenv_s(name, value);  // NOLINT
+    assert(r == 0);
+  } else {
+#endif
+  r = uv_os_setenv(name, value);
+#ifdef WIN32
+  }
+#endif
```

Comment:
Doing `uv_os_setenv` always, would avoid some `#ifdef` noise, and avoids the `else{` case. It's harmless--and it keeps `envmap` "up to date", which may avoid confusion.